### ISSUE TITLE
Some advices on function improving for ccxt and pro

### ISSUE THREAD

**Issue by zhangxf55** [ccxt/ccxt#15188]
- Binance market symbols do not follow naming conventions strictly. For example, Most of the exchanges name a swap trading symbol as BTC/USDT:USDT, but binance names a swap(actually, binance call swap 'future') trading symbol as BTC/USDT, without the suffix :USDT
- While watching an orderbook(or ticker etc...) with ccxt.pro, why not expose an callback function while the orderbook changed? This will improve scalability a lot, while we can write the data into a third party storage/cache, such as redis, mongo or mysql for 
further analysis.

**Comment by carlosmiei** [comment]
Hello @zhangxf55 

> Binance market symbols do not follow naming conventions strictly. For example, Most of the exchanges name a swap trading symbol as BTC/USDT:USDT, but binance names a swap(actually, binance call swap 'future') trading symbol as BTC/USDT, without the suffix :USDT

Yes binance is one of the only exchanges that does not have yet symbols unified; we're working on it hopefully will be fixed soon

> While watching an orderbook(or ticker etc...) with ccxt.pro, why not expose an callback function while the orderbook changed? 
Thanks for your suggestion; we will take it into consideration

**Comment by ttodua** [comment]
Yes, we plan to implement subscribe/unsubscribe event-driver approach in near future.

**Comment by goznauk** [comment]
Any updates?

**Comment by ttodua** [comment]
binance symbols were unified in the past, so the problem should be gone.

probably within a month we will be releasing event-driver approach so you can monitor incoming updates easily. (keen an eye in [`announcements` channel](https://docs.ccxt.com/#/README:~:text=Announcement%20channels%3A)).

**Comment by ttodua** [comment]
let us know if anyone faces any issues.
